eng/common/pipelines/templates/archetype-typespec-emitter.yml

```
@@ -202,16 +202,19 @@ extends:
                 Write-Host "Using overrides.json to generate emitter-package.json"
                 tsp-client generate-config-files `
                 --package-json '$(buildArtifactsPath)/lock-files/package.json' `
-                --output-dir '$(Build.ArtifactStagingDirectory)' `
-                --emitter-package-json-path  '${{ parameters.EmitterPackageJsonPath }}' `
-                --overrides '$(buildArtifactsPath)/overrides.json'
+                --output-dir '$(Build.SourcesDirectory)' `
+                --emitter-package-json-path "$(Build.SourcesDirectory)/${{ parameters.EmitterPackageJsonPath }}" `
+                --overrides '$(buildArtifactsPath)/overrides.json' `
+                --version '$(emitterVersion)'
```

```suggestion
                --overrides '$(buildArtifactsPath)/overrides.json' 
```

`--version` isn't a parameter to `generate-config-files`. It should be extracted from `--package-json`

---

This is a parameter in generate-config-files. The issue I was hitting is that the version extracted from package.json was the base version and didn't include the suffix.

---

It is set for this job and here is the successful pipeline run - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4745993&view=results

---

I guess $(emitterVersion) is essentially a global available to all steps?

---

I would double check this as I don't see where this value is set in this context.